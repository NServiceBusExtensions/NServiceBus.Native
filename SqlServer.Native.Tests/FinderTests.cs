using System;
using System.Collections.Generic;
using System.Text;
using ObjectApproval;
using SqlServer.Native;
using Xunit;

public class FinderTests
{
    static DateTime dateTime = new DateTime(2000, 1, 1, 1, 1, 1, DateTimeKind.Utc);

    static FinderTests()
    {
        DbSetup.Setup();
    }

    [Fact]
    public void FinderSingle()
    {
        SqlHelpers.Drop(Connection.ConnectionString, "FinderTests").Await();
        QueueCreator.Create(Connection.ConnectionString, "FinderTests").Await();
        var sender = new Sender("FinderTests");

        var message = BuildMessage("00000000-0000-0000-0000-000000000001");
        sender.Send(Connection.ConnectionString, message).Await();
        var finder = new Finder("FinderTests");
        var received = finder.Find(Connection.ConnectionString, 1).Result;
        ObjectApprover.VerifyWithJson(received);
    }

    [Fact]
    public void FinderBatch()
    {
        SqlHelpers.Drop(Connection.ConnectionString, "FinderTests").Await();
        QueueCreator.Create(Connection.ConnectionString, "FinderTests").Await();
        var sender = new Sender("FinderTests");

        sender.Send(
            Connection.ConnectionString,
            new List<Message>
            {
                BuildMessage("00000000-0000-0000-0000-000000000001"),
                BuildMessage("00000000-0000-0000-0000-000000000002"),
                BuildMessage("00000000-0000-0000-0000-000000000003"),
                BuildMessage("00000000-0000-0000-0000-000000000004"),
                BuildMessage("00000000-0000-0000-0000-000000000005")
            }).Await();

        var finder = new Finder("FinderTests");
        var messages = new List<Message>();
        finder.Find(
            connection: Connection.ConnectionString,
            size: 3,
            startRowVersion: 2,
            action: message => { messages.Add(message); })
            .Await();
        ObjectApprover.VerifyWithJson(messages);
    }

    static Message BuildMessage(string guid)
    {
        return new Message(new Guid(guid), "theCorrelationId", "theReplyToAddress", dateTime, "headers", Encoding.UTF8.GetBytes("{}"));
    }
}