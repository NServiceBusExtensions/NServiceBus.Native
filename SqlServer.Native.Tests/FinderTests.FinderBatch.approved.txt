[
  {
    "Id": "00000000-0000-0000-0000-000000000002",
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Expires": "2000-01-01T01:01:01",
    "Headers": "headers",
    "Body": "e30="
  },
  {
    "Id": "00000000-0000-0000-0000-000000000003",
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Expires": "2000-01-01T01:01:01",
    "Headers": "headers",
    "Body": "e30="
  },
  {
    "Id": "00000000-0000-0000-0000-000000000004",
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Expires": "2000-01-01T01:01:01",
    "Headers": "headers",
    "Body": "e30="
  }
]